### PR TITLE
move diffract.Constraint to util.Constraint

### LINKED ISSUE #185: Constraint class belongs in util module
A new capability (to report software versions) belongs in `hkl.util` but there is an `import` cycle problem discovered by unit testing: `util` imports `diffract` imports `calc` imports `sample` imports `util`

The `util` module should not import any other _hklpy_ module, otherwise it is not truly a utility.  The `Constraint` class has to move into `util` from `diffract`, which will break the above `import` cycle.

_Originally posted by @prjemian in https://github.com/bluesky/hklpy/issues/160#issuecomment-864308796_

### PR DESCRIPTION
**Pull request by prjemian**
FIX #185 to avoid an import cycle as described in the issue

Make some big notes in the change history since the existing examples and docs point to the old location.

### CODE REVIEW

**Comment by prjemian** [comment]
Actually, since `hkl.diffract` imports `from hkl.util import Constraint`, existing code and documentation will continue to work as described.  PR #184 will provide shortcuts so that `from hkl import Constraint` will be the simpler way, and this is described in the documentation and examples for that PR.  No more work to do here.  _And_, all tests pass now.

@mrakitin -- ready for review.

**Comment by prjemian** [comment]
Note: PR #160 is waiting on these changes.

File: hkl/diffract.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 Common Support for diffractometers
 
 .. autosummary::
-
```
Comment (mrakitin) [inline hkl/diffract.py:8]: Don't we need an extra blank line here for sphinx to render it correctly?

File: hkl/util.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 Utility functions and structures.
 
 .. autosummary::
+    ~Constraint
```
Comment (mrakitin) [inline hkl/util.py:5]: Same comment about the blank line here.

**Review by mrakitin — COMMENTED** [review]
Looks good, have a small concern about the sphinx formatting.

File: hkl/diffract.py (hunk 1 above)
Comment (prjemian) [reply hkl/diffract.py:8]: Good catch.

File: hkl/util.py (hunk 2 above)
Comment (prjemian) [reply hkl/util.py:5]: op cit.

Comment (prjemian) [reply hkl/util.py:6]: found one in `user.py` also

**Comment by prjemian** [comment]
@mrakitin - All changes in.

**Review by mrakitin — APPROVED** [review]

**Comment by prjemian** [comment]
@mrakitin Many thanks!

**Pull request merged**